### PR TITLE
Feat/login screen

### PR DESCRIPTION
**Pull request by kenny-phan-goldenowl**
# Task:
* Update ui for splash, register, login screens
# Description:
* Update ui
* Update router, app styles, coordinator, etc
* Add some widgets for global use
* Remove some unused code from tempalate
# Media showcase
https://github.com/user-attachments/assets/5245d41c-5693-4ab1-873c-c4ef72f44f42

### CODE REVIEW

File: lib/src/features/login/view/login_view.dart [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                'Welcome Back',
+                style: TextStyle(
+                  fontFamily: 'Poppins',
+                  fontWeight: FontWeight.bold,
+                  fontSize: 18,
+                  height: 1.5,
+                ),
+              ),
+              const SizedBox(height: 30),
+              SvgPicture.asset('assets/svgs/login_screen_bg.svg',
```
Comment (krish-do-goldenowl) [inline lib/src/features/login/view/login_view.dart]: Use Flutter gen package for using assets => https://pub.dev/packages/flutter_gen

File: lib/src/features/splashScreen/view/splash_screen_view.dart [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return BaseLayout(
+      buttonText: 'Get started',
+      onButtonPressed: () => AppCoordinator.showRegisterScreen(),
+      content: Column(
+        children: [
+          SvgPicture.asset('assets/svgs/splash_screen_bg.svg',
+              semanticsLabel: "Splash screen bg image", width: 200),
```
Comment (krish-do-goldenowl) [inline lib/src/features/splashScreen/view/splash_screen_view.dart:20]: what is this `semanticsLabel`?

File: lib/src/features/login/view/login_view.dart [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    return BaseLayout(
+      buttonText: 'Login',
+      onButtonPressed: submitForm,
+      content: Column(
+        children: [
+          Column(
+            children: [
+              const Text(
+                'Welcome Back',
+                style: TextStyle(
```
Comment (krish-do-goldenowl) [inline lib/src/features/login/view/login_view.dart]: Get this style from `lib/src/theme/styles.dart` to reduce the duplicated code

File: lib/widgets/layout/base_layout.dart [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+  const BaseLayout({
+    super.key,
+    required this.buttonText,
+    required this.onButtonPressed,
+    required this.content,
+    this.currentView,
+  });
+
+  void _onSignInPressed() {
```
Comment (krish-do-goldenowl) [inline lib/widgets/layout/base_layout.dart]: Do not do the specific logic into a base class => use onTapPress callback and do the logic where the instance declared

File: lib/widgets/layout/base_layout.dart [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+import 'package:myapp/src/router/coordinator.dart';
+import 'package:myapp/src/theme/styles.dart';
+import 'package:myapp/widgets/button/custom_button.dart';
+
+class BaseLayout extends StatelessWidget {
+  final String buttonText;
+  final VoidCallback onButtonPressed;
+  final Widget content;
+  final String? currentView;
```
Comment (krish-do-goldenowl) [inline lib/widgets/layout/base_layout.dart]: Do not use String for specific type => use `enum` instead

File: lib/widgets/layout/base_layout.dart [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+                ),
+                content,
+                Column(
+                  children: [
+                    CustomButton(
+                      title: buttonText,
+                      onButtonPressed: onButtonPressed,
+                    ),
+                    const SizedBox(height: 30),
+                    if (currentView == 'register') ...[
```
Comment (krish-do-goldenowl) [inline lib/widgets/layout/base_layout.dart]: Do not hardcode the type by String => we have to use the `neum` for the specific type when using the OOP language, please note.

**Review by krish-do-goldenowl — CHANGES_REQUESTED** [review]

File: lib/src/features/login/view/login_view.dart (hunk 1 above)
Comment (krish-do-goldenowl) [reply lib/src/features/login/view/login_view.dart]: Please check for all places

File: lib/src/features/login/view/login_view.dart (hunk 3 above)
Comment (krish-do-goldenowl) [reply lib/src/features/login/view/login_view.dart]: Please check for all places

File: lib/src/features/splashScreen/view/splash_screen_view.dart (hunk 2 above)
Comment (kenny-phan-goldenowl) [reply lib/src/features/splashScreen/view/splash_screen_view.dart:20]: it the same with `alt` prop from image tag in html, i follow the guide from SvgPicture

**Review by krish-do-goldenowl — APPROVED** [review]

**Pull request merged**